src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewModel.swift

```diff
@@ -15,13 +15,24 @@ class ExposureDetectionViewModel: CountdownTimerDelegate {
 
 	init(
 		homeState: HomeState,
-		onInactiveButtonTap: @escaping (@escaping (ExposureNotificationError?) -> Void) -> Void,
-		onSurveyTap: @escaping () -> Void
+		appConfigurationProvider: AppConfigurationProviding,
+		onSurveyTap: @escaping (String?) -> Void,
+		onInactiveButtonTap: @escaping (@escaping (ExposureNotificationError?) -> Void) -> Void
 	) {
 		self.homeState = homeState
+		self.appConfigurationProvider = appConfigurationProvider
 		self.onInactiveButtonTap = onInactiveButtonTap
 		self.onSurveyTap = onSurveyTap
 
+		appConfigurationProvider.appConfiguration()
+			.sink { [weak self] in
```

Comment:
Please correct me, I'm not very experienced in combine. But I think this sink is only executed once. Since it is not a `published property` or a `CurrentValueSubject`. There is a Future behind this call, which is resolved once.
With this in mind, the `appConfiguration` state can get stale. I would propose to have the `appConfigurationProvider` as a property, and request the app-config every time the high-risk-model is generated.

Reply:
I'm not very experienced with it too, but right now the way to get the actual appconfiguration has to be done by using a sink, i mean you cant access the properties directly from the appConfigurationProvider like the screenshot
<img width="559" alt="Screenshot 2021-02-03 at 15 30 05" src="https://user-images.githubusercontent.com/15270737/106761113-ca1a8400-6634-11eb-9359-62773a717225.png">